azurelinuxagent/distro/suse/rdma.py

```diff
@@ -0,0 +1,72 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2014 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
+import glob
+import os
+import azurelinuxagent.logger as logger
+from azurelinuxagent.distro.default.rdma import RDMAHandler
+
+
+class SUSERDMAHandler(RDMAHandler):
+    def install_driver(self):
+        """Install the appropropriate driver package for the RDMA firmware"""
+
+        fw_version = self._RDMAHandler__get_rdma_version()
+        zypper_install = 'zypper -n in'
+        zypper_search = 'zypper se -s'
+        package_name = 'msft-rdma-kmp-default'
+        cmd = '%s %s' % (zypper_search, package_name)
+        repo_package_info = os.popen(cmd).readlines()
+        for entry in repo_package_info:
+            if package_name in entry:
+                installed = entry.split('|')[0].strip()
+                version = entry.split('|')[3].strip()
+                if fw_version in version:
+                    if installed == 'i':
+                        info_msg = 'Driver package "%s-%s" '
+                        info_msg += 'is already installed, nothing to do'
+                        logger.info(info_msg % (package_name, version))
+                        break
+                    cmd = '%s %s-%s' % (zypper_install, package_name, version)
+                    result = os.system(cmd)
+                    if result:
+                        error_msg = 'Failed install of package "%s-%s" '
+                        error_msg += 'from available repositories.'
+                        logger.error(error_msg % (package_name, version))
+                    break
+        else:
```

Comment:
On 06/09/2016 03:57 PM, Ahmet Alp Balkan wrote:
> In azurelinuxagent/distro/suse/rdma.py
> <https://github.com/Azure/WALinuxAgent/pull/227#discussion_r66510693>:
> 
>> +                installed = entry.split('|')[0].strip()
>> +                version = entry.split('|')[3].strip()
>> +                if fw_version in version:
>> +                    if installed == 'i':
>> +                        info_msg = 'Driver package "%s-%s" '
>> +                        info_msg += 'is already installed, nothing to do'
>> +                        logger.info(info_msg % (package_name, version))
>> +                        break
>> +                    cmd = '%s %s-%s' % (zypper_install, package_name, version)
>> +                    result = os.system(cmd)
>> +                    if result:
>> +                        error_msg = 'Failed install of package "%s-%s" '
>> +                        error_msg += 'from available repositories.'
>> +                        logger.error(error_msg % (package_name, version))
>> +                    break
>> +        else:
> 
> trying to find |if| of this |else|... 😃

Its the |else| for the |for| loop, no if to be found ;)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/WALinuxAgent/pull/227/files/8a98437b35295b51cd0a394cd6e4d6709d73b70b#r66510693>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABCWz1YQKL0KpVH_yzWua5StBzRMhdfHks5qKHAjgaJpZM4Ix6OA>.
> 

-- 
Robert Schweikert                   MAY THE SOURCE BE WITH YOU
Public Cloud Architect                         LINUX
rjschwei@suse.com
IRC: robjo